internal/v1/handler_compose_image.go

```diff
@@ -495,17 +493,23 @@ func (h *Handlers) buildTemplateRepositories(ctx echo.Context, templateID string
 		return nil, nil, nil, fmt.Errorf("Template %v has no repositories", templateID)
 	}
 
-	for _, id := range *template.RepositoryUuids {
-		// Check if repo is Red Hat
-		rhRepoMap, err = h.server.csClient.GetRepositories(ctx.Request().Context(), []string{}, []string{id}, false)
-		if err != nil {
-			return nil, nil, nil, fmt.Errorf("Unable to retrieve Red Hat repositories: %v", err)
+	rhRepoMap, err := h.server.csClient.GetRepositories(ctx.Request().Context(), []string{}, *template.RepositoryUuids, false)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("Unable to retrieve Red Hat repositories: %v", err)
+	}
+	if len(rhRepoMap) > 0 {
```

Comment:
Technically a length check isn't necessary here, `for repoID := range rhRepoMap ...` will already do what you want and just not iterate

Reply:
oh yes, fixed :)